extensions/namespace-lookup/src/test/java/io/druid/server/namespace/cache/JDBCExtractionNamespaceTest.java

```diff
@@ -120,80 +152,130 @@ public JDBCExtractionNamespaceTest(
+  private void insertValues(final String key, final String val, final String updateTs) throws InterruptedException
   {
-    dbi.withHandle(
-        new HandleCallback<Void>()
-        {
-          @Override
-          public Void withHandle(Handle handle) throws Exception
-          {
-            final String query;
-            if (tsColumn == null) {
-              handle.createStatement(
-                  String.format("DELETE FROM %s WHERE %s='%s'", tableName, keyName, key)
-              ).execute();
-              handle.commit();
-              query = String.format(
-                  "INSERT INTO %s (%s, %s) VALUES ('%s', '%s')",
-                  tableName,
-                  keyName, valName,
-                  key, val
-              );
-            } else {
-              query = String.format(
-                  "INSERT INTO %s (%s, %s, %s) VALUES ('%s', '%s', '%s')",
-                  tableName,
-                  tsColumn, keyName, valName,
-                  updateTs, key, val
-              );
-            }
-            if (1 != handle.createStatement(query).execute()) {
-              throw new ISE("Did not return the correct number of rows");
+    final CountDownLatch committed = new CountDownLatch(1);
+    Assert.assertNull(
+        dbi.withHandle(
+            new HandleCallback<Void>()
+            {
+              @Override
+              public Void withHandle(Handle handle) throws Exception
+              {
+                try {
+                  final String query;
+                  if (tsColumn == null) {
+                    handle.createStatement(
+                        String.format("DELETE FROM %s WHERE %s='%s'", tableName, keyName, key)
+                    ).execute();
+                    handle.commit();
+                    query = String.format(
+                        "INSERT INTO %s (%s, %s) VALUES ('%s', '%s')",
+                        tableName,
+                        keyName, valName,
+                        key, val
+                    );
+                  } else {
+                    query = String.format(
+                        "INSERT INTO %s (%s, %s, %s) VALUES ('%s', '%s', '%s')",
+                        tableName,
+                        tsColumn, keyName, valName,
+                        updateTs, key, val
+                    );
+                  }
+                  if (1 != handle.createStatement(query).execute()) {
+                    throw new ISE("Did not return the correct number of rows");
+                  }
+                  handle.commit();
+                  return null;
+                }
+                finally {
+                  committed.countDown();
+                }
+              }
             }
-            handle.commit();
-            return null;
-          }
-        }
+        )
     );
+    committed.await(10, TimeUnit.SECONDS);
+    Thread.sleep(2);
```

Comment:
If latch completes within 1 ms it makes sure a few ms have elapsed. This is to help make sure things which only check on granularity of ms have a few ms spacing to acknowledge timestamp changes..